agent/config/types.go

```diff
@@ -210,3 +219,42 @@ func (data *SensitiveRawMessage) UnmarshalJSON(jsonData []byte) error {
 	data.contents = json.RawMessage(jsonData)
 	return nil
 }
+
+// DefaultBool makes it possible to understand if a variable was set explicitly or relies on a default setting
+type DefaultBool int
+
+// Enabled is a convenience function for when consumers don't care if the value is implicit or explicit
+func (b DefaultBool) Enabled() bool {
+	return b == ExplicitlyEnabled || b == DefaultEnabled
+}
+
+func (b DefaultBool) MarshalJSON() ([]byte, error) {
+	switch b {
+	case ExplicitlyEnabled:
+		return json.Marshal(true)
+	case ExplicitlyDisabled:
+		return json.Marshal(false)
+	default:
+		return json.Marshal(nil)
+	}
+}
+
+func (b *DefaultBool) UnmarshalJSON(jsonData []byte) error {
+	jsonString := string(jsonData)
+	jsonBool, err := strconv.ParseBool(jsonString)
+
+	switch {
+	case jsonString == "" || jsonString == "null":
```

Comment:
When would the `null` case happen?

Reply:
`null` happens if the struct is serialized to json without `omitempty`. Note that this is the json `null` value and not a json string `"null"`.

I don't know if the first case `""` can happen in practice -- I'll double check